src/main/drivers/bus_i2c_stm32f30x.c

```diff
@@ -231,15 +231,17 @@ bool i2cWrite(uint8_t addr_, uint8_t reg, uint8_t data)
         }
     }
 
-    /* Send Register address */
-    I2C_SendData(I2Cx, (uint8_t) reg);
+    if (reg != -1) {
```

Comment:
OK, thanks for testing.
The code works because it calls next `I2C_TransferHandling` unconditionally (without sending configured byte).
Flags are different (TXIS is sent until NBYTES is transferred, TCR is not sent), but ignored.

The code is OK ... and I2C needs major redesign anyway ...